plugin/session_view.py

```diff
@@ -379,5 +380,27 @@ def _on_code_lenses_resolved_async(self, mode: str) -> None:
     def get_resolved_code_lenses_for_region(self, region: sublime.Region) -> Generator[CodeLens, None, None]:
         yield from self._code_lenses.get_resolved_code_lenses_for_region(region)
 
+    # --- textDocument/inlayHint ---------------------------------------------------------------------------------------
+
+    def present_inlay_hints_async(self, phantoms: List[sublime.Phantom]) -> None:
+        self._inlay_hints_phantom_set.update(phantoms)
+
+    def remove_inlay_hint_phantom(self, phantom_uuid: str) -> None:
+        for p in self._inlay_hints_phantom_set.phantoms:
+            print(getattr(p, 'lsp_uuid'))
+
+        new_phantoms = list(filter(
+            lambda p: getattr(p, 'lsp_uuid') != phantom_uuid,
+            self._inlay_hints_phantom_set.phantoms)
+        )
+        self._inlay_hints_phantom_set.update(new_phantoms)
+
+    def remove_all_inlay_hints(self) -> None:
+        if not self._inlay_hints_phantom_set.phantoms:
+            return
```

Comment:
I added that because `remove_all_inlay_hints` will be called on each document change, if TemporarySettings.SHOW_INLAY_HINTS is False.

So for me updating phantoms looked more expensive than having this check here.
if but will remove this check.

Reply:
[113fbb9](https://github.com/sublimelsp/LSP/pull/2018/commits/113fbb9a47b858268c0306df7f0d41bf74f810a9)

Reply:
toggling was removed so this is not relevant anymore